samples/packages/awscdk-typescript/src/construct/relations.ts

```diff
@@ -1,15 +1,69 @@
+import { FargateService, FargateTaskDefinition } from "aws-cdk-lib/aws-ecs";
+import { Key } from "aws-cdk-lib/aws-kms";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { Topic } from "aws-cdk-lib/aws-sns";
+import { SqsSubscription } from "aws-cdk-lib/aws-sns-subscriptions";
+import { Queue } from "aws-cdk-lib/aws-sqs";
 import { Construct } from "constructs";
-import { PackageInfraConfig } from "../package-config";
+import { PackageInfraConfig, RelationTyping } from "../package-config";
+
+interface RelationsPropsFromComponents {
+  keys: { [key: string]: Key },
+  queues: { [key: string]: Queue },
+  topics: { [key: string]: Topic },
+  buckets: { [key: string]: Bucket },
+}
+
+interface RelationsPropsFromServices {
+  taskdefs: { [key: string]: FargateTaskDefinition },
+  services: { [key: string]: FargateService },
+}
 
 export interface RelationsProps {
   config: PackageInfraConfig,
+  components: RelationsPropsFromComponents,
+  services: RelationsPropsFromServices,
```

Comment:
A typescript trick used here to make possible to reuse the props from `Services` and `Components` in `PackagePlanner`